### ISSUE TITLE
Have to log in to Xbox Live on xboxapi.com every few hours

### ISSUE THREAD

**Issue by iliketoprogram14** [xboxapi/Xbox-One-API-Issues#38]
Like the title says, I have to log into Xbox Live every few hours (I think every 8 hours); otherwise, whenever I invoke an API function, I get the "must refresh tokens".  Is this expected behavior, to manually refresh tokens every few hours?  If so, is there any way to automate this?

Some background info on my account: I have to log in using Oauth as I use two factor authentication for my MSA and app passwords don't seem to work with the normal username/password method.  My account is on the free tier.

Let me know if there's anything I can do or provide to get this working.

**Comment by djekl** [comment]
Unfortunately if you use the oAuth logins, these expire faster than the standard method and can't be refreshed. This is an issue I am still yet to solve, problems when your using something thats undocumented and shouldn't have lol.

This isn't anything to do with the tier of your account, its an issue with the oAuth tokens from Microsoft. Unless you require access to account specific information such as messages, then the only other thing than removing 2FA is to create a scraper account, and save the password with us.

**Comment by iliketoprogram14** [comment]
Ah, that's too bad.  I also tried [app passwords](https://support.microsoft.com/en-us/help/12409/microsoft-account-app-passwords-two-step-verification), but for whatever reason I couldn't even login with that, which is dumb on Microsoft's part.

I ended up creating a scraper account as you recommended, but one really cool side effect of the API that I'm missing out on is that I'm no longer able to view my primary account's real online presence, regardless of whether or not I have chosen to appear online or offline.  Not a big deal though.

Thanks for responding!
